prov/gni/include/gnix_freelist.h

```diff
@@ -123,6 +126,16 @@ static inline int _gnix_fl_alloc(struct dlist_entry **e, struct gnix_freelist *f
         fastlock_acquire(&fl->lock);
 
     if (dlist_empty(&fl->freelist)) {
+
+        if (fl->refill_size == 0) {
+                ret = -FI_ECANCELED;
+
+                GNIX_DEBUG(FI_LOG_DEBUG, "Unable to grow freelist(%p) since the"
+                        " refill size is %d\n", fl, fl->refill_size);
```

Comment:
Sorry to nitpick, but it's not that you're *unable* to grow the free list, but this is the policy of the freelist implementation.  I'd just have it say something like" Freelist not growable (refill size == 0)".